src/shared/ui/BaseComponent.ts

```diff
@@ -0,0 +1,61 @@
+export abstract class BaseComponent<S = unknown> {
+  protected $root: Element | null;
```

Comment:
`$root`가 querySelector의 반환 타입만 가진다면 JavaScript가 내부적으로 제공해주는 타입을 활용하는게 추후 유지보수 측면에서 더 좋을 것 같다는 생각을 했는데요. querySelector의 타입이 어느 순간에 휙 바뀌어 버릴수가 있는데, 그렇게 된다면 개발자가 일일이 찾아서 변경해줘야 할 것 같아요.

```suggestion
  protected $root: ReturnType<ParentNode['querySelector']>;
```

이런식으로 자바스크립트가 제공해주는 타입에 의존한다면, 개발자는 자바스크립트 변경사항과는 무관하게 타입을 수정할 필요가 없어질 것 같아요.

정원님은 어떻게 생각하시는지 의견을 듣고 싶어요!

Reply:
너무 좋은 의견이예요!

그런데 저는 동규님께서 [이 코맨트](https://github.com/hanghae-plus/front_3rd_chatper1/pull/16/files#r1777212067)에서 남겨주신 '타입을 공통화'하는 방법이 더 마음에 들어요 ㅎㅎ

```suggestion
  protected $root: RootElement;
```

이렇게요! 😆


왜냐하면 유틸리티 타입에 인터페이스를 중첩해서 사용하는 것이 어쩌면 조금 불필요하게 복잡해 보일 수도 있을 것 같았어요.

그래서 저는 동규님께서 다른 코멘트에서 제안해주신 것처럼, 타입을 공통화하는 방식이 더 좋아보이는데, 동규님은 어떻게 생각하시나요?

여러 가지 방법으로 제안해주신 덕분에 여러가지 측면에서 생각해보게 되었어요.
정말 감사합니다!! 🙇‍♀️🙇‍♀️